File: .NET/VS2010TrainingKit/Labs/AspNetWebForms4/Source/Ex04-ViewState/begin/C#/WebFormsSampleApp/Models/AdventureWorksRepository.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

using System.Linq;
using System.Data.Linq;

namespace WebFormsSampleApp.Models
{
    public class AdventureWorksRepository
    {
        public ProductCategory[] GetCategories()
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return (from c in db.ProductCategories
                    where c.ParentProductCategoryID == null
                    select c).ToArray();
        }

        public ProductCategory[] GetSubcategories(string category)
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();

            return (from c in db.ProductCategories
                    where c.ParentProductCategoryID == GetCategoryId(category)
                    select c).ToArray();
        }

        public int GetProductsCount()
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return db.Products.Count();
        }

        public int GetProductsCountByCategories(int[] categoryIds)
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return (from p in db.Products
                    where categoryIds.Contains(p.ProductCategoryID.Value)
                    select p).Count();
        }

        public Product[] GetProductsByCategories(int[] categoryIds, int page, int pageSize)
        {
            int startRowIndex = (page-1) * pageSize;
            int maximumRows = pageSize;
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return (from p in db.Products
                    where categoryIds.Contains(p.ProductCategoryID.Value)
                    orderby p.Name ascending
                    select p)
                    .Skip(startRowIndex)
                    .Take(maximumRows)
                    .ToArray();
        }

        public byte[] GetProductImage(int productId)
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return (from p in db.Products
                    where p.ProductID == productId
                    select p.ThumbNailPhoto).SingleOrDefault<Binary>().ToArray();
        }

        public Product GetProduct(int productId)
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            return db.Products.Where(p => p.ProductID == productId).SingleOrDefault();
        }

        private int GetCategoryId(string category)
        {
            AdventureWorksDataContext db = new AdventureWorksDataContext();
            var result = (from c in db.ProductCategories
                          where c.Name == category
                          select c.ProductCategoryID);

            return (result.Count<int>() > 0) ? result.First<int>() : -1;
        }
    }
}

File: .NET/VS2010TrainingKit/Labs/AspNetWebForms4/Source/Ex02-ClientId/begin/VB/WebFormsSampleApp/Models/AdventureWorksRepository.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System.Linq
Imports System.Data.Linq

Namespace WebFormsSampleApp.Models
	Public Class AdventureWorksRepository
		Public Function GetCategories() As ProductCategory()
			Dim db As New AdventureWorksDataContext()
			Return ( _
			    From c In db.ProductCategories _
			    Where c.ParentProductCategoryID Is Nothing _
			    Select c).ToArray()
		End Function

		Public Function GetSubcategories(ByVal category As String) As ProductCategory()
			Dim db As New AdventureWorksDataContext()

			Return ( _
			    From c In db.ProductCategories _
			    Where c.ParentProductCategoryID.Equals(GetCategoryId(category)) _
			    Select c).ToArray()
		End Function

		Public Function GetProductsCount() As Integer
			Dim db As New AdventureWorksDataContext()
			Return db.Products.Count()
		End Function

		Public Function GetProductsCountByCategories(ByVal categoryIds() As Integer) As Integer
			Dim db As New AdventureWorksDataContext()
			Return ( _
			    From p In db.Products _
			    Where categoryIds.Contains(p.ProductCategoryID.Value) _
			    Select p).Count()
		End Function

		Public Function GetProductsByCategories(ByVal categoryIds() As Integer, ByVal page As Integer, ByVal pageSize As Integer) As Product()
			Dim startRowIndex As Integer = (page-1) * pageSize
			Dim maximumRows As Integer = pageSize
			Dim db As New AdventureWorksDataContext()
			Return ( _
			    From p In db.Products _
			    Where categoryIds.Contains(p.ProductCategoryID.Value) _
			    Order By p.Name Ascending _
			    Select p).Skip(startRowIndex).Take(maximumRows).ToArray()
		End Function

		Public Function GetProductImage(ByVal productId As Integer) As Byte()
			Dim db As New AdventureWorksDataContext()
            Return ( _
                From p In db.Products _
                Where p.ProductID = productId _
                Select p.ThumbNailPhoto).SingleOrDefault().ToArray()
		End Function

		Public Function GetProduct(ByVal productId As Integer) As Product
			Dim db As New AdventureWorksDataContext()
			Return db.Products.Where(Function(p) p.ProductID = productId).SingleOrDefault()
		End Function

		Private Function GetCategoryId(ByVal category As String) As Integer
			Dim db As New AdventureWorksDataContext()
			Dim result = ( _
			    From c In db.ProductCategories _
			    Where c.Name = category _
			    Select c.ProductCategoryID)

            Return If((result.Count() > 0), result.First(), -1)
		End Function
	End Class
End Namespace
